src/cmake_modules/FastDownwardMacros.cmake

```diff
@@ -125,79 +101,54 @@ function(fast_downward_add_existing_sources_to_list _SOURCES_LIST_VAR)
+    cmake_parse_arguments(_LIBRARY "${_OPTIONS}" "${_ONE_VALUE_ARGS}" "${_MULTI_VALUE_ARGS}" ${ARGN})
     # Check mandatory arguments.
-    if(NOT _PLUGIN_NAME)
-        message(FATAL_ERROR "fast_downward_plugin: 'NAME' argument required.")
-    endif()
-    if(NOT _PLUGIN_SOURCES)
-        message(FATAL_ERROR "fast_downward_plugin: 'SOURCES' argument required.")
-    endif()
-    fast_downward_add_existing_sources_to_list(_PLUGIN_SOURCES)
-    # Check optional arguments.
-    if(NOT _PLUGIN_DISPLAY_NAME)
-        string(TOLOWER ${_PLUGIN_NAME} _PLUGIN_DISPLAY_NAME)
-    endif()
-    if(NOT _PLUGIN_HELP)
-        set(_PLUGIN_HELP ${_PLUGIN_DISPLAY_NAME})
+    if(NOT _LIBRARY_NAME)
+        message(FATAL_ERROR "fast_downward_library: 'NAME' argument required.")
     endif()
-    # Decide whether the plugin should be enabled by default.
-    if (DISABLE_PLUGINS_BY_DEFAULT)
-        set(_OPTION_DEFAULT FALSE)
-    else()
-        set(_OPTION_DEFAULT TRUE)
-    endif()
-    # Overwrite default value for core plugins and dependecy-only plugins.
-    if (_PLUGIN_CORE_PLUGIN)
-        set(_OPTION_DEFAULT TRUE)
-    elseif(_PLUGIN_DEPENDENCY_ONLY)
-        set(_OPTION_DEFAULT FALSE)
-    endif()
-
-    option(PLUGIN_${_PLUGIN_NAME}_ENABLED ${_PLUGIN_HELP} ${_OPTION_DEFAULT})
-    if(_PLUGIN_DEPENDENCY_ONLY OR _PLUGIN_CORE_PLUGIN)
-        mark_as_advanced(PLUGIN_${_PLUGIN_NAME}_ENABLED)
+    if(NOT _LIBRARY_SOURCES)
+        message(FATAL_ERROR "fast_downward_library: 'SOURCES' argument required.")
     endif()
 
-    set(PLUGIN_${_PLUGIN_NAME}_DISPLAY_NAME ${_PLUGIN_DISPLAY_NAME} PARENT_SCOPE)
-    set(PLUGIN_${_PLUGIN_NAME}_SOURCES ${_PLUGIN_SOURCES} PARENT_SCOPE)
-    set(PLUGIN_${_PLUGIN_NAME}_DEPENDS ${_PLUGIN_DEPENDS} PARENT_SCOPE)
-    list(APPEND PLUGINS ${_PLUGIN_NAME})
-    set(PLUGINS ${PLUGINS} PARENT_SCOPE)
-endfunction()
+    fast_downward_add_existing_sources_to_list(_LIBRARY_SOURCES)
 
-function(fast_downward_add_plugin_sources _SOURCES_LIST_VAR)
-    set(_UNCHECKED_PLUGINS)
-    foreach(PLUGIN ${PLUGINS})
-        if(PLUGIN_${PLUGIN}_ENABLED)
-            list(APPEND _UNCHECKED_PLUGINS ${PLUGIN})
+    if (NOT _LIBRARY_CORE_LIBRARY AND NOT _LIBRARY_DEPENDENCY_ONLY)
+        # Decide whether the plugin should be enabled by default.
+        if (DISABLE_LIBRARIES_BY_DEFAULT)
+            set(_OPTION_DEFAULT FALSE)
+        else()
+            set(_OPTION_DEFAULT TRUE)
         endif()
+        option(LIBRARY_${_LIBRARY_NAME}_ENABLED ${_LIBRARY_HELP} ${_OPTION_DEFAULT})
+    endif()
+
+    add_library(downward_${_LIBRARY_NAME} INTERFACE)
+    target_link_libraries(downward_${_LIBRARY_NAME} INTERFACE cxx_options)
+    target_sources(downward_${_LIBRARY_NAME} INTERFACE ${_LIBRARY_SOURCES})
+    foreach(DEPENDENCY ${_LIBRARY_DEPENDS})
+        target_link_libraries(downward_${_LIBRARY_NAME} INTERFACE downward_${DEPENDENCY})
     endforeach()
+    # TODO: This feels very hacky. I'd rather have the ability to specify
```

Comment:
We could just move the code to `DownwardFiles.cmake` without a special option. Something like this:

```CMake
create_fast_downward_library(
    NAME UTILS
    HELP "System utilities"
    SOURCES
        # ...
        utils/timer
    CORE_PLUGIN
    CORE_LIBRARY
)

if(UNIX AND NOT APPLE)
    target_link_libraries(downward_UTILS INTERFACE rt)
endif()

# On Windows, find the psapi library for determining peak memory.
if(WIN32)
    cmake_policy(SET CMP0074 NEW)
    target_link_libraries(downward_UTILS INTERFACE psapi)
endif()
```

Reply:
done